File: jksoa-rpc/jksoa-rpc-client/src/test/resources/ISimpleService.php
<?php
namespace net\jkcode\jksoa\rpc\example;

class ISimpleService {

	// ------------ 方法注解 ------------
	public static $_methodAnnotations = [];

	// ------------ php对java调用映射的方法 ------------
	public static function hostname(){
		return 'java.lang.String hostname()';
	}

	public static function sayHi($msg){
		return 'java.lang.String sayHi(java.lang.String)';
	}

	public static function sleep(){
		return 'long sleep()';
	}

	public static function checkVersion(){
		return 'void checkVersion()';
	}

	public static function testException(){
		return 'void testException()';
	}

	// ------------ 降级方法 ------------
}

File: jksoa-rpc/jksoa-rpc-client/src/main/kotlin/net/jkcode/jksoa/rpc/example/ISimpleService.kt
package net.jkcode.jksoa.rpc.example

import co.paralleluniverse.fibers.Suspendable
import net.jkcode.jksoa.common.annotation.RemoteService
import java.rmi.RemoteException

/**
 * 简单示例的服务接口
 *
 * @author shijianhang
 * @create 2017-12-15 下午7:37
 **/
@RemoteService(version = 1, onlyLeader = true)
interface ISimpleService /*: Remote // rmi协议服务接口 */ {

    @Throws(RemoteException::class) // rmi异常
    @Suspendable
    fun hostname(): String

    @Throws(RemoteException::class) // rmi异常
    fun sayHi(name: String = "shi"): String

    @Throws(RemoteException::class) // rim异常
    fun checkVersion()

    @Throws(RemoteException::class) // rim异常
    @JvmDefault
    fun defaultMethod(msg:String){
        println("call default method, with parameter: $msg")
    }

    /**
     * 抛个异常
     */
    @Throws(RemoteException::class) // rim异常
    fun testException()

}